sim/src/generic_eps_hardware_model.cpp

```diff
@@ -104,219 +185,295 @@ namespace Nos3
         else if (command.compare("DISABLE") == 0) 
         {
             _enabled = GENERIC_EPS_SIM_ERROR;
-            _count = 0;
-            _config = 0;
-            _status = 0;
             response = "Generic_epsHardwareModel::command_callback:  Disabled";
         }
-        else if (command.substr(0,7).compare("STATUS=") == 0)
-        {
-            try
-            {
-                _status = std::stod(command.substr(7));
-                response = "Generic_epsHardwareModel::command_callback:  Status set";
-            }
-            catch (...)
-            {
-                response = "Generic_epsHardwareModel::command_callback:  Status invalid";
-            }            
-        }
         else if (command.compare("STOP") == 0) 
         {
             _keep_running = false;
             response = "Generic_epsHardwareModel::command_callback:  Stopping";
         }
-        /* TODO: Add anything additional commands here */
 
         /* Send a reply */
         sim_logger->info("Generic_epsHardwareModel::command_callback:  Sending reply: %s.", response.c_str());
         _command_node->send_reply_message_async(msg, response.size(), response.c_str());
     }
 
-
-    /* Custom function to prepare the Generic_eps HK telemetry */
-    void Generic_epsHardwareModel::create_generic_eps_hk(std::vector<uint8_t>& out_data)
+    void Generic_epsHardwareModel::eps_switch_update(const std::uint8_t sw_num, uint8_t sw_status)
     {
-        /* Prepare data size */
-        out_data.resize(16, 0x00);
+        /* Is the switch valid? */
+        if (sw_num < 8)
+        {
+            /* Is the status valid? */
+            if ((sw_status == 0x00) || (sw_status == 0xAA))
+            {
+                /* Prepare the simulator bus with the switch node */
```

Comment:
Delete this and the next line... dead, useless code

Reply:
I think printing in the log that an invalid state write occurred is useful